src/Shouldly/MessageGenerators/ShouldBeSubsetOfMessageGenerator.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Shouldly.MessageGenerators
+{
+    internal class ShouldBeSubsetOfMessageGenerator : ShouldlyMessageGenerator
+    {
+        private static readonly Regex Validator = new Regex("ShouldBeSubsetOf", RegexOptions.Compiled);
+
+        public override bool CanProcess(ShouldlyAssertionContext context)
+        {
+            return Validator.IsMatch(context.ShouldMethod);
+        }
+
+        public override string GenerateErrorMessage(ShouldlyAssertionContext context)
+        {
+            const string format = @"{0} should be subset of {1} but {2} do not";
```

Comment:
Maybe one of the following instead (better English)

 - `{0} should be subset of {1} but {2} are outside subset`
 - `{0} should be subset of {1} but {2} are not`
 - `{0} should be subset of {1} but {2} are additional`

Not quite sure which is best. @robdmoore @xerxesb any thoughts?